apps/learning-snippets/src/test/property-grid/Telemetry.test.tsx

```diff
@@ -0,0 +1,140 @@
+import { ECSchemaRpcImpl } from "@itwin/ecschema-rpcinterface-impl";
+import { PresentationRpcInterface } from "@itwin/presentation-common";
+import { HierarchyCacheMode, initialize as initializePresentationTesting, terminate as terminatePresentationTesting } from "@itwin/presentation-testing";
+// __PUBLISH_EXTRACT_START__ PropertyGrid.ComponentWithTelemetryImports
+import { PropertyGridComponent } from "@itwin/property-grid-react";
+import { UiItemsManager } from "@itwin/appui-react";
+// __PUBLISH_EXTRACT_END__
+// __PUBLISH_EXTRACT_START__ PropertyGrid.ComponentWithTelemetryWrapperImports
+import {  PropertyGrid, TelemetryContextProvider } from "@itwin/property-grid-react";
+// __PUBLISH_EXTRACT_END__
+import { cleanup, render, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { buildIModel, insertPhysicalElement, insertPhysicalModelWithPartition, insertSpatialCategory } from "../../utils/IModelUtils";
+import { PropertyGridTestUtils } from "../../utils/PropertyGridTestUtils";
+
+describe("Property grid", () => {
+  describe("Learning snippets", () => {
+    describe("Telemetry", () => {
+      before(async function () {
+        await initializePresentationTesting({
+          backendProps: {
+            caching: {
+              hierarchies: {
+                mode: HierarchyCacheMode.Memory,
+              },
+            },
+          },
+          testOutputDir: join(__dirname, "output"),
+          backendHostProps: {
+            cacheDir: join(__dirname, "cache"),
+          },
+          rpcs: [SnapshotIModelRpcInterface, IModelReadRpcInterface, PresentationRpcInterface, ECSchemaRpcInterface],
+        });
+        // eslint-disable-next-line @itwin/no-internal
+        ECSchemaRpcImpl.register();
+      });
+
+      after(async function () {
+        await terminatePresentationTesting();
+      });
+
+      beforeEach(async () => {
+        await PropertyGridTestUtils.initialize();
+      });
+
+      afterEach(async () => {
+        await PropertyGridTestUtils.terminate();
+        sinon.restore();
+        // eslint-disable-next-line @itwin/no-internal
+        UiItemsManager.clearAllProviders();
+      });
+
+      it("Renders component with feature usage and performance tracking", async function () {
+        const imodelConnection = (
+          await buildIModel(this, async (builder) => {
+            const physicalModel = insertPhysicalModelWithPartition({ builder, codeValue: "TestPhysicalModel" });
+            const category = insertSpatialCategory({ builder, codeValue: "Test SpatialCategory" });
+            insertPhysicalElement({ builder, modelId: physicalModel.id, categoryId: category.id });
+            return { category };
+          })
+        ).imodel;
+        sinon.stub(UiFramework, "getIModelConnection").returns(imodelConnection);
+        const consoleSpy = sinon.spy(console, "log");
```

Comment:
I think it would be better to pass spy or stub to onPerformanceMeasured and check arguments it was invoked with instead of calling console.log.

Reply:
Changed it to spies, the functions have the same name as the properties in context or `PropertyGridComponent` do you think it's clear enough that these should be the user's functions and not something we provide?

Reply:
Good point. I forgot that these will be included in Readme. Maybe in that case something like this could work
```ts
onFeatureUsed={(featureId) => {
  // log feature usage 
  logUsage(featureId);
}}
```

Or maybe instead of `logUsage` your could do as before but instead of calling `console.log` just create `log` stub and call it.

Reply:
Adjusted.